## Import Libraries

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report, accuracy_score, confusion_matrix

## Data Loading and Preparation

In [2]:
df = pd.read_csv("data.csv")
# Select features and target variable
X = df[['tempo', 'beats', 'chroma_stft', 'rmse', 'spectral_centroid', 'spectral_bandwidth', 'rolloff', 'zero_crossing_rate']]
y = df['label']
# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [3]:
# Standardize the features
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

#use kernel='poly' to transform dataset !
model = SVC(kernel='poly', degree=1,  gamma='scale')
#  change
model.fit(X_train_scaled, y_train)

SVC(degree=1, kernel='poly')

In [4]:
# Make predictions on the test set
y_pred = model.predict(X_test_scaled)
y_pred 

array(['disco', 'pop', 'pop', 'metal', 'pop', 'metal', 'metal', 'jazz',
       'metal', 'blues', 'rock', 'jazz', 'metal', 'rock', 'reggae',
       'reggae', 'blues', 'disco', 'pop', 'disco', 'blues', 'blues',
       'hiphop', 'jazz', 'rock', 'country', 'classical', 'blues', 'metal',
       'rock', 'reggae', 'classical', 'metal', 'metal', 'metal',
       'classical', 'disco', 'disco', 'jazz', 'blues', 'disco', 'jazz',
       'metal', 'disco', 'country', 'blues', 'country', 'metal', 'hiphop',
       'jazz', 'jazz', 'hiphop', 'blues', 'jazz', 'metal', 'reggae',
       'metal', 'blues', 'pop', 'blues', 'jazz', 'jazz', 'jazz',
       'classical', 'metal', 'metal', 'metal', 'disco', 'blues',
       'classical', 'classical', 'metal', 'classical', 'pop', 'country',
       'blues', 'disco', 'country', 'jazz', 'disco', 'rock', 'blues',
       'jazz', 'rock', 'disco', 'jazz', 'blues', 'country', 'pop', 'rock',
       'rock', 'metal', 'blues', 'blues', 'country', 'hiphop', 'hiphop',
       'jazz',

In [5]:
# Evaluate the model
accuracy = accuracy_score(y_test, y_pred)
accuracy

0.44

The SVM model with a polynomial kernel (of degree 1) achieved an accuracy of 44% on the test dataset. This indicates that the model correctly predicted the label for 44% of the instances in the test set.